### PR TITLE
Botocore updates

### PR DESCRIPTION
**Pull request by thehesiod**
- move AioConfig to new file like in botocore
- implement AioConfig.merge like in botocore
- AioClientCreator._get_client_args use as much as parent's version as
possible
- update calls to new botocore signatures

(runtime weakref issue resolved, comment added to client.py so others don't run into the same issue)

### CODE REVIEW

**Comment by jettify** [comment]
Cool! Botocore moving fast this days:)  Have you tried to execute tests? you can do that with
```
py.test tests/
```

File: setup.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from setuptools import setup, find_packages
 
 
-install_requires = ['botocore==1.3.26', 'aiohttp>=0.21.2']
+install_requires = ['botocore', 'aiohttp>=0.21.2']
```
Comment (jettify) [inline setup.py]: lets do `botocore>=1.4.0` here

**Comment by thehesiod** [comment]
botocore.exceptions.ClientError: An error occurred (NoSuchBucket) when calling the ListObjects operation: The specified bucket does not exist
botocore.exceptions.ClientError: An error occurred (AccessDenied) when calling the CreateBucket operation: Access Denied

**Comment by jettify** [comment]
may be eventual consistency issue?

**Comment by thehesiod** [comment]
oh, I need perms to be able to create buckets?  For my work location I don't have permissions.  I'll try with my personal account

**Comment by jettify** [comment]
I am  executing tests too

**Comment by thehesiod** [comment]
looks like maybe pagination is broken?  let me c what changed

**Comment by jettify** [comment]
Yeah i have two failed tests:
test_paginate_max_items
test_paginate_within_page_boundaries

**Comment by jettify** [comment]
looks like I fixed that issue. hold on a sec

**Comment by jettify** [comment]
I only managed to fix `test_paginate_max_items` with patch:
```
diff --git a/aiobotocore/paginate.py b/aiobotocore/paginate.py
index 595f5cd..c8b81f8 100644
--- a/aiobotocore/paginate.py
+++ b/aiobotocore/paginate.py
@@ -54,7 +54,7 @@ class AioPageIterator(PageIterator):
         self._is_stop = False
         self._current_kwargs = self._op_kwargs
         self._previous_next_token = None
-        self._next_token = [None for _ in range(len(self._input_token))]
+        self._next_token = dict((key, None) for key in self._input_token)
         # The number of items from result_key we've seen so far.
         self._total_items = 0
         self._first_request = True
```

**Comment by thehesiod** [comment]
cool, the signature changed a bit, let me c if I can get it to look like botocore

**Comment by jettify** [comment]
have a fix for other test

**Comment by thehesiod** [comment]
cool!

**Comment by jettify** [comment]
Other fix 
```
diff --git a/aiobotocore/paginate.py b/aiobotocore/paginate.py
index 595f5cd..d33eb8f 100644
--- a/aiobotocore/paginate.py
+++ b/aiobotocore/paginate.py
@@ -29,24 +31,6 @@ class AioPageIterator(PageIterator):

         self._init_pager()

-    @property
-    def result_keys(self):
-        return self._result_keys
-
-    @property
-    def resume_token(self):
-        """Token to specify to resume pagination."""
-        return self._resume_token
-
-    @resume_token.setter
-    def resume_token(self, value):
-        if isinstance(value, list):
-            self._resume_token = '___'.join([str(v) for v in value])
-
-    @property
-    def non_aggregate_part(self):
-        return self._non_aggregate_part
-
```
just delete those properties, I do not remember why I left those...

**Comment by jettify** [comment]
I executed tests one more time, test suite passed :)

**Comment by jettify** [comment]
Thank you for your PR!

**Comment by thehesiod** [comment]
my pleasure! thanks for your help!

**Pull request merged**